### ISSUE TITLE
`joinstyle` is not respected for `FancyArrowPatch` (either the path or the arrow heads)

### ISSUE THREAD

**Issue by david-hoffman** [matplotlib/matplotlib#13641]
### Bug report

**Bug summary**

`joinstyle` is not respected for `FancyArrowPatch` (either the path or the arrow heads)

**Code for reproduction**

```python
import matplotlib as mpl
import matplotlib.pyplot as plt

# high dpi so you can see the difference
fig, ax = plt.subplots(dpi=600)

# iterate through styles
for i, js in enumerate(("miter", "round", "bevel"), start=1):
    arrow_head = mpl.patches.ArrowStyle.CurveFilledAB(head_length=4, head_width=2)
    connection = mpl.patches.ConnectionStyle.Bar(fraction=0, angle=-90)
    double_arrow = mpl.patches.FancyArrowPatch((0.1 * i, 1 - 0.1), (1 - 0.1, 0.1 * i), linewidth=2,
                                       arrowstyle=arrow_head, connectionstyle=connection, joinstyle=js)
    ax.add_patch(double_arrow)
```

**Actual outcome**

![image](https://user-images.githubusercontent.com/3699983/54078050-77d95880-428f-11e9-964e-1c85b211d254.png)

**Expected outcome**

Different join styles for each line (including arrow heads)

**Matplotlib version**
  * Operating system: Windows 10
  * Matplotlib version: 3.0.2
  * Matplotlib backend (`print(matplotlib.get_backend())`): Qt5Agg
  * Python version: 
```
Python 3.7.2 (default, Feb 21 2019, 17:35:59) [MSC v.1915 64 bit (AMD64)]
Type 'copyright', 'credits' or 'license' for more information
IPython 7.2.0 -- An enhanced Interactive Python. Type '?' for help.
```

I installed with the default `conda` channel.

**Comment by ImportanceOfBeingErnest** [comment]
The joinstyle and capstyle are fixed for `FancyArrowPatch`. 

https://github.com/matplotlib/matplotlib/blob/21f1355a94cd164f80530bf410439066dbe2bb99/lib/matplotlib/patches.py#L4287-L4289

I hostly don't know why that is the case. If we remove that line the result would look like this:

![image](https://user-images.githubusercontent.com/23121882/54091770-3c6c8600-4384-11e9-969c-14ac2490e16f.png)

Is that what you're expecting? Would someone know why that got fixed in the first place?

**Comment by ImportanceOfBeingErnest** [comment]
Ok, [here is the deal](https://github.com/matplotlib/matplotlib/pull/11741#discussion_r205949368) (in a comment by @anntzer):

>  Yes, ignoring the capstyle was probably a long-standing bug [..] but changing that would result in changes in the baseline images [..].  I'll leave this as it is until someone actually complains or we have some othe reason to change the baseline images.

Actually, I don't quite see why we can't keep `"round"` as default - in that way no baseline images need to change and it remains backwards compatible - but at the same time allow for user input to change it.

**Comment by ImportanceOfBeingErnest** [comment]
A possible fix is at #13647, no changes are made to the defaults and tests pass as they are.
